http/nodepeers.go

```diff
@@ -0,0 +1,59 @@
+package http
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/attestantio/go-eth2-client/api"
+	apiv1 "github.com/attestantio/go-eth2-client/api/v1"
+	"strings"
+)
+
+type peersResponseJSON struct {
+	Data []apiv1.Peer           `json:"data"`
+	Meta map[string]interface{} `json:"meta,omitempty"`
+}
+
+func (s *Service) NodePeers(ctx context.Context, opts *api.PeerOpts) (*api.Response[*apiv1.Peers], error) {
+	var peerResonse peersResponseJSON
+	// all options are considered optional
+	request := "/eth/v1/node/peers"
+	var additionalFields []string
+
+	for _, stateFilter := range opts.State {
+		additionalFields = append(additionalFields, fmt.Sprintf("state=%s", stateFilter))
+	}
+
+	for _, directionFilter := range opts.Direction {
+		additionalFields = append(additionalFields, fmt.Sprintf("direction=%s", directionFilter))
+	}
+
+	if len(additionalFields) > 0 {
+		request += "?"
+		for _, field := range additionalFields {
+			request += fmt.Sprintf("%s&", field)
+		}
+	}
+	//remove the trailing '&'
+	request = strings.TrimSuffix(request, "&")
```

Comment:
This is quite involved; would be easier to do a single `request = fmt.Sprintf("%s?%s", request, strings.Join(additionalFields,"&"))`